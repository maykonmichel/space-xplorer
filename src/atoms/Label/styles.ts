import {StyleSheet} from 'react-native';

import colors from '~/theme/colors';

export default StyleSheet.create({
  text: {
    color: colors.primary.dark,
    fontSize: 16,
    fontWeight: 'bold',
    textTransform: 'uppercase',
  },
});
