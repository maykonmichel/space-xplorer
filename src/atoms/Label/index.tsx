import React, {FC} from 'react';
import {Text, TextProps} from 'react-native';

import styles from './styles';

export type Props = TextProps;

const Label: FC<Props> = ({children, style, ...props}) => (
  <Text style={[styles.text, style]} {...props}>
    {children}
  </Text>
);

export default Label;
