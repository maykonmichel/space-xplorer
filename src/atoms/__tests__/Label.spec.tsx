import {render} from '@testing-library/react-native';
import React from 'react';

import Label from '~/atoms/Label';

describe('<Label />', () => {
  it('should render as expected', () => {
    const {toJSON} = render(<Label>Some label</Label>);

    expect(toJSON()).toMatchSnapshot();
  });
});
