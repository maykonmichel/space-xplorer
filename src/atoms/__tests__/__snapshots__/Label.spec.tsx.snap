// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render as expected 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#B61827",
        "fontSize": 16,
        "fontWeight": "bold",
        "textTransform": "uppercase",
      },
      undefined,
    ]
  }
>
  Some label
</Text>
`;
